src/pudl/transform/ferc1.py

```diff
@@ -2923,6 +2924,123 @@ def transform(
     }
 
 
+def ferc1_transform_asset_factory(
+    table_name: str, ferc1_tfr_classes: dict[str:Ferc1AbstractTableTransformer]
+) -> AssetsDefinition:
+    """Create an asset that pulls in raw ferc Form 1 assets and applies transformations.
+
+    This is a convenient way to create assets for tables that only depend on raw dbf,
+    raw xbrl instant and duration tables and xbrl metadata. For tables with additional
+    upsream dependencies, create a stand alone asset using an asset decorator. See
+    the plants_steam_ferc1 asset.
+
+    Args:
+        table_name: The name of the table to create an asset for.
+        ferc1_tfr_classes: A dictionary of table names to the corresponding transformer class.
+
+    Return:
+        An asset for the clean table.
+    """
+    ins = {}
+    ins["raw_dbf"] = AssetIn(TABLE_NAME_MAP[table_name]["dbf"])
+    ins["raw_xbrl_instant"] = AssetIn(TABLE_NAME_MAP[table_name]["xbrl"] + "_instant")
+    ins["raw_xbrl_duration"] = AssetIn(TABLE_NAME_MAP[table_name]["xbrl"] + "_duration")
+    ins["xbrl_metadata_json"] = AssetIn("xbrl_metadata_json")
+
+    tfr_class = ferc1_tfr_classes[table_name]
+
+    @asset(name=table_name, ins=ins, io_manager_key="pudl_sqlite_io_manager")
+    def ferc1_transform_asset(
+        raw_dbf: pd.DataFrame,
+        raw_xbrl_instant: pd.DataFrame,
+        raw_xbrl_duration: pd.DataFrame,
+        xbrl_metadata_json: dict[str, dict[str, list[dict[str, Any]]]],
+    ) -> pd.DataFrame:
+        """Transform a FERC Form 1 table.
+
+        Args:
+            raw_dbf: raw dbf table.d
+            raw_xbrl_instant: raw XBRL instant table.
+            raw_xbrl_duration: raw XBRL duration table.
+            xbrl_metadata_json: XBRL metadata json for all tables.
+
+        Returns:
+            transformed FERC Form 1 table.
+        """
+        df = tfr_class(xbrl_metadata_json=xbrl_metadata_json[table_name]).transform(
+            raw_dbf=raw_dbf,
+            raw_xbrl_instant=raw_xbrl_instant,
+            raw_xbrl_duration=raw_xbrl_duration,
+        )
+        return convert_cols_dtypes(df, data_source="ferc1")
+
+    return ferc1_transform_asset
+
+
+def create_ferc1_transform_assets() -> list[AssetsDefinition]:
+    """Create a list of transformed FERC Form 1 assets.
+
+    Returns:
+        A list of AssetsDefinitions where each asset is a clean ferc form 1 table.
+    """
+    ferc1_tfr_classes = {
```

Comment:
This duplicates a mapping of transformers in the `transform` function - are we planning on keeping the top-level `transform` function around after we port things to Dagster? Looks like the EIA files don't have a corresponding top-level `transform` function anymore.

Reply:
Yes, this would replace the top-level `transform` function.